src/PIL/Image.py

```diff
@@ -3069,7 +3069,15 @@ def frombuffer(mode, size, data, decoder_name="raw", *args):
     return frombytes(mode, size, data, decoder_name, args)
 
 
-def fromarray(obj, mode=None):
+class SupportsArrayInterface(Protocol):
```

Comment:
Since (I assume) this is not intended to be exported, I would expect we either want this to be private, i.e.
```suggestion
class _SupportsArrayInterface(Protocol):
```
or in a private module, i.e. `PIL._typing.SupportsArrayInterface`.

@radarhere what do you think?

Reply:
I was modelling this after our implementation of [`SupportsGetMesh`](https://pillow.readthedocs.io/en/stable/reference/ImageOps.html#PIL.ImageOps.SupportsGetMesh).

https://github.com/python-pillow/Pillow/pull/7812#issuecomment-1952150296
> If we're using _SupportsGetMesh as a type hint, this probably is something users might want to use, so should be accessible as a non-underscore thing?

Reply:
I feel like the difference is that `SupportsGetMesh` is a Pillow protocol, whereas `SupportsArrayInterface` seems to be a numpy protocol. Similarly, we have [`PIL._typing.SupportsRead`](https://pillow.readthedocs.io/en/stable/reference/internal_modules.html#PIL._typing.SupportsRead) for a builtin Python protocol. I would therefore expect users to either use the actual numpy type (e.g. `numpy.ndarray`) or define their own protocol.

@hugovk Do you have any thoughts as the author of the quoted comment?

Reply:
Hmm, let's go for `SupportsArrayInterface`, it's probably more useful that way, and it's only a small class. We an always deprecate/remove later if needed.